backend/django/deploy-to-aws.md

```diff
@@ -0,0 +1,157 @@
+    - In "10_packages.config":
+      ```
+      packages:
+        yum:
+          git: []
+          postgresql95-devel: []
+          libjpeg-turbo-devel: []
+          libffi: []
+          libffi-devel: []
+      ```
+    - Copy "20_elasticcache.config", "30_options.config", "50_apache.config" and "enable_mod_deflate.conf" as is
+    - Copy "40_python.config" but change the last two lines with:
+      ```
+      "aws:elasticbeanstalk:container:python:staticfiles":
+        "/static/": "/staticfiles/"
+      ```
+- To connect to RDS, add a database configuration to your Django config in "base.py" or "production.py":
+  ```python3
+  # DATABASE CONFIGURATION
+  # ------------------------------------------------------------------------------
+
+  if 'RDS_DB_NAME' in os.environ:
+    DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql_psycopg2',
+            'NAME': env('RDS_DB_NAME'),
+            'USER': env('RDS_USERNAME'),
+            'PASSWORD': env('RDS_PASSWORD'),
+            'HOST': env('RDS_HOSTNAME'),
+            'PORT': env('RDS_PORT'),
+        }
+    }
+  else:
+      DATABASES['default'] = env.db('DATABASE_URL')
+  ```
+- To serve your statics from AWS, update also the Django config in "base.py":
+  ```python3
+  # STATIC FILE CONFIGURATION
+  # ------------------------------------------------------------------------------
+
+  STATIC_ROOT = 'staticfiles'
+  STATIC_URL = '/staticfiles/'
+  ```
+- You can commit your setup
+  ```bash
+  git add .
+  git commit -m "Added python, django and database config and requirements files"
+  ```
+
+### Deploying the application
+- For the first time you will create the instance, hence all the necessary servers:
+  ```bash
+  eb create --scale 1 -db -db.engine postgres -db.i db.t2.micro
+  ```
+  - Name your environment, like:  project-name-staging
+  - Select the default CNAME
+  - Select the classic load balancer option
+  - Choose a username and password to connect to the DB
+- EB will now generate all your servers and provision your config then launch your project
+- But you might still have errors `eb logs`, you are missing environment variables that you can directly add to AWS website
+  - On AWS website, on the top right, select the right server location (ex: London)
+  - Go to Elastic Beanstalk
+  - Enter your new application
+  - Go to configuration
+  - Modify the software configuration with variables such as:
+    ```
+    DJANGO_ACCOUNT_ALLOW_REGISTRATION, DJANGO_ADMIN_URL, DJANGO_ALLOWED_HOSTS, DJANGO_AWS_ACCESS_KEY_ID, DJANGO_AWS_SECRET_ACCESS_KEY, DJANGO_MAILGUN_API_KEY, DJANGO_SECRET_KEY, DJANGO_SECURE_SSL_REDIRECT, DJANGO_SENTRY_DSN, DJANGO_SERVER_EMAIL, DJANGO_SETTINGS_MODULE, MAILGUN_SENDER_DOMAIN, REDIS_ENDPOINT_ADDRESS, REDIS_PORT
+    ```
+- Check: You can verify on the AWS website that each server peace has been created (EB, RDS, S3, EC2)
+
+### Create a superadmin
+- TODO
```

Comment:
Now that you have deployed once, on your host you should have the credentials for AWS (`cat ~/.aws/config`). 
To create a super user you will have to connect directly to the machine in order to run ./manage.py createsuperuser.

To ssh to your EC2 machine run `eb ssh [PUT YOUR INSTANCE NAME IF NOT BY DEFAULT]` 

Then on the machine go to the root of the project:
-`cd /opt/python/current`
Add the python environment:
- `source ./env`
Run the createuser command
- `cd /app`
- `./manage.py createsuperuser`